### DISCUSSION TITLE
Hot Reload includes handling certificate rotation

### DISCUSSION THREAD

**Discussion by mp3monster** [fluent/fluent-bit#8120] [Ideas]
Im told that refreshing certificates for SSL for the HTTP plugin would require Fluent Bit to be restarted. It would be good if the certificate could be hot reloaded to avoid FLB down time.
It would make it a lot easier to manage frequent certificate rotation such as the case of operating with Let'sEncrypt which deliberately provides short lived certs. or operating environments with gitops.

**Comment by patrick-stephens** [comment]
@mp3monster Phil, Hiroshi (@cosmo0920) indicates it should work with hot reload so if it doesn't then that's a bug.

More generally I think reloading certs would be helpful, particularly with the increasing usage of cert rotation with shorter duration certs. It may be worthwhile doing that on a failure just in case it is down to cert rotation. However that's probably more of a feature request.

**Comment by sonman** [comment]
In 3.0.7 `kill SIGHUP` does not reload certificates. At least strace does not show any read on file configure with **tls.ca_file**

**Reply by patrick-stephens** [reply]
3.0.7 is fairly old now, I would update to latest.
